qln/signtx.go

```diff
@@ -148,70 +151,152 @@ func (nd *LitNode) SignClaimTx(claimTx *wire.MsgTx, value int64, pre []byte,
 }
 
 // SignNextState generates your signature for their state.
-func (nd *LitNode) SignState(q *Qchan) ([64]byte, error) {
-
+func (nd *LitNode) SignState(q *Qchan) ([64]byte, [][64]byte, error) {
 	var sig [64]byte
 
 	// make sure channel exists, and wallet is present on node
 	if q == nil {
-		return sig, fmt.Errorf("SignState nil channel")
+		return sig, nil, fmt.Errorf("SignState nil channel")
 	}
 	_, ok := nd.SubWallet[q.Coin()]
 	if !ok {
-		return sig, fmt.Errorf("SignState no wallet for cointype %d", q.Coin())
+		return sig, nil, fmt.Errorf("SignState no wallet for cointype %d", q.Coin())
 	}
 	// build transaction for next state
-	tx, err := q.BuildStateTx(false) // their tx, as I'm signing
+	commitmentTx, spendHTLCTxs, HTLCTxOuts, err := q.BuildStateTxs(false) // their tx, as I'm signing
 	if err != nil {
-		return sig, err
+		return sig, nil, err
 	}
 
 	// make hash cache for this tx
-	hCache := txscript.NewTxSigHashes(tx)
+	hCache := txscript.NewTxSigHashes(commitmentTx)
 
 	// generate script preimage (ignore key order)
 	pre, _, err := lnutil.FundTxScript(q.MyPub, q.TheirPub)
 	if err != nil {
-		return sig, err
+		return sig, nil, err
 	}
 
 	// get private signing key
 	priv, err := nd.SubWallet[q.Coin()].GetPriv(q.KeyGen)
 	if err != nil {
-		return sig, err
+		return sig, nil, err
 	}
 
 	// generate sig.
 	bigSig, err := txscript.RawTxInWitnessSignature(
-		tx, hCache, 0, q.Value, pre, txscript.SigHashAll, priv)
+		commitmentTx, hCache, 0, q.Value, pre, txscript.SigHashAll, priv)
 	// truncate sig (last byte is sighash type, always sighashAll)
 	bigSig = bigSig[:len(bigSig)-1]
 
 	sig, err = sig64.SigCompress(bigSig)
 	if err != nil {
-		return sig, err
+		return sig, nil, err
 	}
 
-	log.Printf("____ sig creation for channel (%d,%d):\n", q.Peer(), q.Idx())
-	log.Printf("\tinput %s\n", tx.TxIn[0].PreviousOutPoint.String())
-	for i, txout := range tx.TxOut {
-		log.Printf("\toutput %d: %x %d\n", i, txout.PkScript, txout.Value)
+	fmt.Printf("____ sig creation for channel (%d,%d):\n", q.Peer(), q.Idx())
```

Comment:
`log.Printf`? If so, maybe just global search. Most of these `fmt` statements were replaced by `log` when you were working on this.